src/Uno.UI.Composition/Composition/Visual.skia.cs

```diff
@@ -609,7 +610,10 @@ private unsafe void CreateLocalSession(in PaintingSession parentSession, out Pai
 #if DEBUG
 		else
 		{
-			Debug.Assert(Unsafe.IsNullRef(ref rootTransform) ? canvas.TotalMatrix == TotalMatrix.ToSKMatrix() : canvas.TotalMatrix == (TotalMatrix * rootTransform).ToSKMatrix());
+			Debug.Assert(Unsafe.IsNullRef(ref rootTransform)
+				? canvas.TotalMatrix == TotalMatrix.ToSKMatrix()
+				// Due to the limit precision of doubles, instead of comparing the two matrices directly we compare the Frobenius norm of their difference to zero
+				: (canvas.TotalMatrix.ToMatrix4x4() - TotalMatrix * rootTransform).ToSKMatrix().Values.Sum(i => i * i) < 1e-5);
```

Comment:
I'm pretty sure the Frobenius approach here is not the right one.

If you're concerned by the performance here, you should consider using SIMD or AVX, btw. In a tested helper, obviously.

Reply:
Why not? I don't really know which way would work best, I used whatever googling gave me. Performance is irrelevant here since it's just a Debug assertion,